src/engine/audio/audio.cpp

```diff
@@ -502,6 +542,52 @@ void AudioManager::DrawSources() {
     plSetMeshUniformColour(mesh, PLColour(255, 0, 0, 255));
 }
 
+void AudioManager::PlayMusic(const std::string &path) {
+    const AudioSample *sample = CacheSample(path);
+    if(sample == nullptr) {
+        return;
+    }
+
+    if(music_source_ == nullptr) {
+        // Setup our global music source
+        const PLConsoleVariable *volume_var = plGetConsoleVariable("audio_volume_music");
+        // todo: this should never happen, instead plGetConsoleVariable should support defaults...
+        if(volume_var == nullptr) {
+            Error("Failed to get \"audio_music_volume\" runtime variable, aborting!\n");
+        }
+        music_source_ = new AudioSource(sample, volume_var->f_value, 1.0f, false);
+    } else {
+        music_source_->StopPlaying();
+        music_source_->SetSample(sample);
+    }
+
+    music_source_->StartPlaying();
+}
+
+void AudioManager::PauseMusic() {
+    if(music_source_ == nullptr) {
+        return;
+    }
+
+    music_source_->Pause();
+}
+
+void AudioManager::StopMusic() {
+    if(music_source_ == nullptr) {
+        return;
+    }
+
+    music_source_->StopPlaying();
+}
+
+void AudioManager::SetMusicVolume(float gain) {
+    music_source_->SetGain(gain);
```

Comment:
Should this check for NULL like above methods?

Reply:
Yup, though really this shouldn't be necessary - there should always be a music source... Something to revisit in future.